### PR TITLE
KT-31301: Escape underscores in package name on auto import

### PR DESCRIPTION
**Pull request by Monchi**
Fixed this: https://youtrack.jetbrains.com/issue/KT-31301

### CODE REVIEW

File: core/descriptors/src/org/jetbrains/kotlin/renderer/RenderingUtils.kt [hunk 1]
```diff
@@ -28,7 +28,7 @@ private fun Name.shouldBeEscaped(): Boolean {
     if (isSpecial) return false
 
     val string = asString()
-    return string in KeywordStringsGenerated.KEYWORDS || string.any { !Character.isLetterOrDigit(it) && it != '_' }
+    return string in KeywordStringsGenerated.KEYWORDS || string.any { !Character.isLetterOrDigit(it) }
```
Comment (dimonchik0036) [inline core/descriptors/src/org/jetbrains/kotlin/renderer/RenderingUtils.kt]: Look at https://teamcity.jetbrains.com/viewLog.html?buildId=2319666
This is incorect fix.

Comment (dimonchik0036) [inline core/descriptors/src/org/jetbrains/kotlin/renderer/RenderingUtils.kt]: Look at https://teamcity.jetbrains.com/viewLog.html?buildId=2319666
This is incorect fix.

Comment (Monchi) [reply core/descriptors/src/org/jetbrains/kotlin/renderer/RenderingUtils.kt]: I see but `gradle ideaPluginTest` failed with latest master at `org.jetbrains.kotlin.kapt3.test.ClassFileToSourceStubConverterTestGenerated`. Is it should succeed?

```
info: [kapt] All Javac options: {-proc:=only, -classpath=/usr/lib/jvm/java-8-openjdk/jre/lib/management-agent.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/jsse.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/jce.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/resources.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/charsets.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/rt.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/zipfs.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/dnsns.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/cldrdata.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/sunjce_provider.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/sunpkcs11.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/jaccess.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/sunec.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/nashorn.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/localedata.jar:/home/monchi/Documents/dev/kotlin/dist/kotlinc/lib/kotlin-stdlib.jar, -processorpath=, -s=/tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_errorLocationMapping/kaptRunner, -d=/tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_errorLocationMapping/kaptRunner, -encoding=UTF-8}
error: /tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_errorLocationMapping/kaptTest/errorLocationMapping.kt:9: エラー: シンボルを見つけられません
class ErrorInConstructorParameter(val a: String, val b: ABC, val c: List<ABC>)
                                                 ^
  シンボル:   クラス ABC
  場所: クラス ErrorInConstructorParameter
error: /tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_errorLocationMapping/kaptTest/errorLocationMapping.kt:9: エラー: シンボルを見つけられません
class ErrorInConstructorParameter(val a: String, val b: ABC, val c: List<ABC>)
                                                             ^
  シンボル:   クラス ABC
  場所: クラス ErrorInConstructorParameter
error: /tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_errorLocationMapping/kaptTest/errorLocationMapping.kt:9: エラー: シンボルを見つけられません
class ErrorInConstructorParameter(val a: String, val b: ABC, val c: List<ABC>)
                                 ^
  シンボル:   クラス ABC
  場所: クラス ErrorInConstructorParameter
error: /tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_errorLocationMapping/kaptTest/errorLocationMapping.kt:11: エラー: シンボルを見つけられません
class ErrorInSupertype : ABC
^
  シンボル: クラス ABC
error: /tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_errorLocationMapping/kaptTest/errorLocationMapping.kt:12: エラー: シンボルを見つけられません
class ErrorInSupertype2 : ABC<String>()
^
  シンボル: クラス ABC
error: /tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_errorLocationMapping/kaptTest/errorLocationMapping.kt:16: エラー: シンボルを見つけられません
    lateinit var p2: ABC
    ^
  シンボル:   クラス ABC
  場所: クラス ErrorInDeclarations
error: /tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_errorLocationMapping/kaptTest/errorLocationMapping.kt:17: エラー: シンボルを見つけられません
    lateinit var p3: BCD<String>
    ^
  シンボル:   クラス BCD
  場所: クラス ErrorInDeclarations
error: /tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_errorLocationMapping/kaptTest/errorLocationMapping.kt:20: エラー: シンボルを見つけられません
    fun overloads(a: ABC) {}
    ^
  シンボル:   クラス ABC
  場所: クラス ErrorInDeclarations
error: /tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_errorLocationMapping/kaptTest/errorLocationMapping.kt:22: エラー: シンボルを見つけられません
    fun f1(a: String, b: ABC<List<String>>) {}
    ^
  シンボル:   クラス ABC
  場所: クラス ErrorInDeclarations
error: /tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_errorLocationMapping/kaptTest/errorLocationMapping.kt:25: エラー: シンボルを見つけられません
    fun f4(): ABC? = null
    ^
  シンボル:   クラス ABC
  場所: クラス ErrorInDeclarations
error: /tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_errorLocationMapping/kaptTest/errorLocationMapping.kt:6: エラー: シンボルを見つけられません
@Anno(ABC::class)
^
  シンボル: クラス ABC


16:12:54,060 DEBUG ij.psi.impl.source.PsiFileImpl - Loaded text for file /tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_errorLocationMapping/kaptTest/errorLocationMapping.kt 

info: [kapt] All Javac options: {-Xmaxerrs=1, -proc:=only, -classpath=/usr/lib/jvm/java-8-openjdk/jre/lib/management-agent.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/jsse.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/jce.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/resources.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/charsets.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/rt.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/zipfs.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/dnsns.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/cldrdata.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/sunjce_provider.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/sunpkcs11.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/jaccess.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/sunec.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/nashorn.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/localedata.jar:/home/monchi/Documents/dev/kotlin/dist/kotlinc/lib/kotlin-stdlib.jar, -processorpath=, -s=/tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_maxErrorCount/kaptRunner, -d=/tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_maxErrorCount/kaptRunner, -encoding=UTF-8}
error: /tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_maxErrorCount/kaptTest/maxErrorCount.kt:8: エラー: シンボルを見つけられません
    fun a(a: ABC, b: BCD) {}
    ^
  シンボル:   クラス ABC
  場所: クラス Test


16:13:10,253 DEBUG ij.psi.impl.source.PsiFileImpl - Loaded text for file /tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_maxErrorCount/kaptTest/maxErrorCount.kt 

info: [kapt] All Javac options: {-proc:=only, -classpath=/usr/lib/jvm/java-8-openjdk/jre/lib/management-agent.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/jsse.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/jce.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/resources.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/charsets.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/rt.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/zipfs.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/dnsns.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/cldrdata.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/sunjce_provider.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/sunpkcs11.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/jaccess.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/sunec.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/nashorn.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/localedata.jar:/home/monchi/Documents/dev/kotlin/dist/kotlinc/lib/kotlin-stdlib.jar, -processorpath=, -s=/tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_importsKt22083/kaptRunner, -d=/tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_importsKt22083/kaptRunner, -encoding=UTF-8}
error: /tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_importsKt22083/kaptTest/c.kt:17: エラー: シンボルを見つけられません
    fun e(): LibFooBar
    ^
  シンボル:   クラス FooBar
  場所: パッケージ lib


16:14:15,102 DEBUG ij.psi.impl.source.PsiFileImpl - Loaded text for file /tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_importsKt22083/kaptTest/a.kt 
16:14:15,104 DEBUG ij.psi.impl.source.PsiFileImpl - Loaded text for file /tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_importsKt22083/kaptTest/b.kt 
16:14:15,105 DEBUG ij.psi.impl.source.PsiFileImpl - Loaded text for file /tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_importsKt22083/kaptTest/c.kt 
16:14:15,131 DEBUG ij.psi.impl.source.PsiFileImpl - Loaded text for file /tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_importsKt22083/java-files/lib/File.java 
16:14:15,132 DEBUG ij.psi.impl.source.PsiFileImpl - Loaded text for file /tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_importsKt22083/java-files/lib/IOException.java 

info: [kapt] All Javac options: {-proc:=only, -classpath=/usr/lib/jvm/java-8-openjdk/jre/lib/management-agent.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/jsse.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/jce.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/resources.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/charsets.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/rt.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/zipfs.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/dnsns.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/cldrdata.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/sunjce_provider.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/sunpkcs11.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/jaccess.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/sunec.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/nashorn.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/localedata.jar:/home/monchi/Documents/dev/kotlin/dist/kotlinc/lib/kotlin-stdlib.jar, -processorpath=, -s=/tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_kt24272/kaptRunner, -d=/tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_kt24272/kaptRunner, -encoding=UTF-8}
error: エラー: Can't generate a stub for 'Foo$Bar$Bar'.
  Its name 'Bar' is the same as one of the outer class names.
  Java forbids it. Please change one of the class names.
error: /tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_kt24272/kaptTest/kt24272.kt:9: エラー: シンボルを見つけられません
        val bars: ArrayList<Bar> = ArrayList()
        ^
  シンボル:   クラス Bar
  場所: クラス Bar


16:14:50,267 DEBUG ij.psi.impl.source.PsiFileImpl - Loaded text for file /tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_kt24272/kaptTest/kt24272.kt 

info: [kapt] All Javac options: {-proc:=only, -classpath=/usr/lib/jvm/java-8-openjdk/jre/lib/management-agent.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/jsse.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/jce.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/resources.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/charsets.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/rt.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/zipfs.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/dnsns.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/cldrdata.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/sunjce_provider.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/sunpkcs11.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/jaccess.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/sunec.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/nashorn.jar:/usr/lib/jvm/java-8-openjdk/jre/lib/ext/localedata.jar:/home/monchi/Documents/dev/kotlin/dist/kotlinc/lib/kotlin-stdlib.jar, -processorpath=, -s=/tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_invalidFieldName/kaptRunner, -d=/tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_invalidFieldName/kaptRunner, -encoding=UTF-8}
error: エラー: 'WHI-TE' is an invalid Java enum value name
error: /tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_invalidFieldName/kaptTest/invalidFieldName.kt:5: エラー: シンボルを見つけられません
@Anno(Color.`WHI-TE`)
^
  シンボル:   変数 InvalidFieldName
  場所: クラス Color


16:15:13,763 DEBUG ij.psi.impl.source.PsiFileImpl - Loaded text for file /tmp/kotlin-annotation-processingProject_test_2910532341697926486/testRootGW5/unitTest_invalidFieldName/kaptTest/invalidFieldName.kt 
```

Comment (dimonchik0036) [reply core/descriptors/src/org/jetbrains/kotlin/renderer/RenderingUtils.kt]: On tests, the main thing is to make sure that we don't have false positives. Try to understand how the auto-import works, maybe the problem is not in this place.

Comment (Monchi) [reply core/descriptors/src/org/jetbrains/kotlin/renderer/RenderingUtils.kt]: Ummm, this is my `gradle ideaPluginTest` report. It seems that there is no fails due to my change.

```
<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
<meta http-equiv="x-ua-compatible" content="IE=edge"/>
<title>Test results - Test Summary</title>
<link href="css/base-style.css" rel="stylesheet" type="text/css"/>
<link href="css/style.css" rel="stylesheet" type="text/css"/>
<script src="js/report.js" type="text/javascript"></script>
</head>
<body>
<div id="content">
<h1>Test Summary</h1>
<div id="summary">
<table>
<tr>
<td>
<div class="summaryGroup">
<table>
<tr>
<td>
<div class="infoBox" id="tests">
<div class="counter">89</div>
<p>tests</p>
</div>
</td>
<td>
<div class="infoBox" id="failures">
<div class="counter">5</div>
<p>failures</p>
</div>
</td>
<td>
<div class="infoBox" id="ignored">
<div class="counter">0</div>
<p>ignored</p>
</div>
</td>
<td>
<div class="infoBox" id="duration">
<div class="counter">4m36.61s</div>
<p>duration</p>
</div>
</td>
</tr>
</table>
</div>
</td>
<td>
<div class="infoBox failures" id="successRate">
<div class="percent">94%</div>
<p>successful</p>
</div>
</td>
</tr>
</table>
</div>
<div id="tabs">
<ul class="tabLinks">
<li>
<a href="#tab0">Failed tests</a>
</li>
<li>
<a href="#tab1">Packages</a>
</li>
<li>
<a href="#tab2">Classes</a>
</li>
</ul>
<div id="tab0" class="tab">
<h2>Failed tests</h2>
<ul class="linkList">
<li>
<a href="classes/org.jetbrains.kotlin.kapt3.test.ClassFileToSourceStubConverterTestGenerated.html">ClassFileToSourceStubConverterTestGenerated</a>.
<a href="classes/org.jetbrains.kotlin.kapt3.test.ClassFileToSourceStubConverterTestGenerated.html#testErrorLocationMapping">testErrorLocationMapping</a>
</li>
<li>
<a href="classes/org.jetbrains.kotlin.kapt3.test.ClassFileToSourceStubConverterTestGenerated.html">ClassFileToSourceStubConverterTestGenerated</a>.
<a href="classes/org.jetbrains.kotlin.kapt3.test.ClassFileToSourceStubConverterTestGenerated.html#testImportsKt22083">testImportsKt22083</a>
</li>
<li>
<a href="classes/org.jetbrains.kotlin.kapt3.test.ClassFileToSourceStubConverterTestGenerated.html">ClassFileToSourceStubConverterTestGenerated</a>.
<a href="classes/org.jetbrains.kotlin.kapt3.test.ClassFileToSourceStubConverterTestGenerated.html#testInvalidFieldName">testInvalidFieldName</a>
</li>
<li>
<a href="classes/org.jetbrains.kotlin.kapt3.test.ClassFileToSourceStubConverterTestGenerated.html">ClassFileToSourceStubConverterTestGenerated</a>.
<a href="classes/org.jetbrains.kotlin.kapt3.test.ClassFileToSourceStubConverterTestGenerated.html#testKt24272">testKt24272</a>
</li>
<li>
<a href="classes/org.jetbrains.kotlin.kapt3.test.ClassFileToSourceStubConverterTestGenerated.html">ClassFileToSourceStubConverterTestGenerated</a>.
<a href="classes/org.jetbrains.kotlin.kapt3.test.ClassFileToSourceStubConverterTestGenerated.html#testMaxErrorCount">testMaxErrorCount</a>
</li>
</ul>
</div>
<div id="tab1" class="tab">
<h2>Packages</h2>
<table>
<thead>
<tr>
<th>Package</th>
<th>Tests</th>
<th>Failures</th>
<th>Ignored</th>
<th>Duration</th>
<th>Success rate</th>
</tr>
</thead>
<tbody>
<tr>
<td class="failures">
<a href="packages/org.jetbrains.kotlin.kapt3.test.html">org.jetbrains.kotlin.kapt3.test</a>
</td>
<td>89</td>
<td>5</td>
<td>0</td>
<td>4m36.61s</td>
<td class="failures">94%</td>
</tr>
</tbody>
</table>
</div>
<div id="tab2" class="tab">
<h2>Classes</h2>
<table>
<thead>
<tr>
<th>Class</th>
<th>Tests</th>
<th>Failures</th>
<th>Ignored</th>
<th>Duration</th>
<th>Success rate</th>
</tr>
</thead>
<tbody>
<tr>
<td class="success">
<a href="classes/org.jetbrains.kotlin.kapt3.test.AbstractClassFileToSourceStubConverterTest.html">org.jetbrains.kotlin.kapt3.test.AbstractClassFileToSourceStubConverterTest</a>
</td>
<td>1</td>
<td>0</td>
<td>0</td>
<td>0.457s</td>
<td class="success">100%</td>
</tr>
<tr>
<td class="failures">
<a href="classes/org.jetbrains.kotlin.kapt3.test.ClassFileToSourceStubConverterTestGenerated.html">org.jetbrains.kotlin.kapt3.test.ClassFileToSourceStubConverterTestGenerated</a>
</td>
<td>78</td>
<td>5</td>
<td>0</td>
<td>3m56.70s</td>
<td class="failures">93%</td>
</tr>
<tr>
<td class="success">
<a href="classes/org.jetbrains.kotlin.kapt3.test.KotlinKapt3IntegrationTests.html">org.jetbrains.kotlin.kapt3.test.KotlinKapt3IntegrationTests</a>
</td>
<td>6</td>
<td>0</td>
<td>0</td>
<td>33.535s</td>
<td class="success">100%</td>
</tr>
<tr>
<td class="success">
<a href="classes/org.jetbrains.kotlin.kapt3.test.KotlinKaptContextTestGenerated.html">org.jetbrains.kotlin.kapt3.test.KotlinKaptContextTestGenerated</a>
</td>
<td>4</td>
<td>0</td>
<td>0</td>
<td>5.909s</td>
<td class="success">100%</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="footer">
<p>
<div>
<label class="hidden" id="label-for-line-wrapping-toggle" for="line-wrapping-toggle">Wrap lines
<input id="line-wrapping-toggle" type="checkbox" autocomplete="off"/>
</label>
</div>Generated by 
<a href="http://www.gradle.org">Gradle 5.1</a> at 2019/06/18 23:28:32</p>
</div>
</div>
</body>
</html>
```

Comment (goodwinnk) [reply core/descriptors/src/org/jetbrains/kotlin/renderer/RenderingUtils.kt]: Thank you for the pull request. Please check tests in `DiagnosticsTestGenerated`. Modifing `shouldBeEscaped()` causes escaping not only in importing but also in other usages, that is unexpected.

I'm closing the PR for now, please reopen it when the tests are fixed.

**Pull request closed without merging**